holoviews/operation/element.py

```diff
@@ -477,6 +478,14 @@ def _process(self, element, key=None):
             ys = ys[:, :-1] + (np.diff(ys, axis=1)/2.)
         data = (xs, ys, zs)
 
+        # if any data is a datetime, transform to matplotlib's numerical format
+        data_is_datetime = tuple(isdatetime(arr) for k, arr in enumerate(data))
+        if any(data_is_datetime):
+            data = tuple(
+                date2num(d) if is_datetime else d
+                for d, is_datetime in zip(data, data_is_datetime)
+            )
+
```

Comment:
I added this explicit conversion to not let QuadContourSet do it internally (in case that behaviour changes some day).